content/blog/express-route-namespace/index.md

```diff
@@ -0,0 +1,233 @@
+  technology and chances are you are going to have to deal with it eventually, at least superficially, in your Front End
+  career. Let's cover the basic concepts and day to day useful commands you will likely use when dealing with Docker."
+tags:
+  - Javascript
+  - Typescript
+  - NodeJs
+  - Express
+  - Namespace
+  - Router
+---
+
+This is going to be a quick post about solving the
+great old question: _"how to create route namespaces with Express"_.
+
+[Show me the answer](TODO)
+
+This isn't at all an advanced technique but we have either very old
+posts talking about how to do it the old way; or we have libraries to
+do it or; we have the Express documentation which rather
+_inexpressive_ if you pardon a pun.
+
+First of all, the Express docs are purely API docs, they tend to not
+cover too much real world scenarios or "recipes" as some call them.
+
+This is specially hard on newcomers that might not have all the
+necessary http, frameworks, nodejs, express idiosyncracy context
+in their heads yet.
+
+So, the closest thing we have to an answer in the Express docs are [here](https://expressjs.com/en/4x/api.html#app.use),
+and say in a very tiny place:
+
+> Since router and app implement the middleware interface, you can use them as you would any other middleware function.
+
+And later goes to actually show the closest thing to an example that
+you will see in the docs:
+
+```javascript
+function mw1(req, res, next) {
+  next()
+}
+function mw2(req, res, next) {
+  next()
+}
+
+var r1 = express.Router()
+r1.get("/", function (req, res, next) {
+  next()
+})
+
+var r2 = express.Router()
+r2.get("/", function (req, res, next) {
+  next()
+})
+
+var subApp = express()
+subApp.get("/", function (req, res, next) {
+  next()
+})
+
+app.use(mw1, [mw2, r1, r2], subApp)
+```
+
+As you can see, this is rather blunt and inexpressive and it is using
+quite old javascript.
+
+The answer is actually right here but you will never
+find it just by googling: _"how to create namespaces in express"_ or
+any other variant.
+
+## The Answer
+
+Although we've seen the answer in a very _expressive_ way in the
+Express docs, here we present it a little bit more unpacked, explained and
+with actual real world usage in mind.
+
+One typical scenario where you would want namespacing is for example
+when you create a Rest api that has all its rest sources under the `/api`.
+
+So for example you would have
+
+- `/api/dog/` -> get all dogs
+- `/api/dog/123` -> get dog with id 123
+- `/api/cat/` -> get all cats
+- `/api/cat/123` -> get cat with id 123
+
+This is nice way of structuring your application since you might have routes
+that are not particularly Rest resources that might go some other place.
+
+Or you might even be working on an app that also servers files and html routes,
+which is very common when building monolithical apps or simply smaller apps that
+don't require any other fancy micro-servicing.
+
+### Enter `Router`
```

Comment:
i think this title (and maybe the prior one) was a little bit confusing because router was introduced in an earlier section